Dockerfile

```diff
@@ -11,21 +11,21 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-ARG image=public.ecr.aws/eks-distro-build-tooling/eks-distro-minimal-base-nonroot:2024-08-13-1723575672.2
-ARG golang_image=public.ecr.aws/docker/library/golang:1.22.5
+ARG image=public.ecr.aws/eks-distro-build-tooling/eks-distro-minimal-base-nonroot:2024-10-01-1727740884.2023
+ARG golang_image=public.ecr.aws/docker/library/golang:1.23.4
 
 FROM --platform=$BUILDPLATFORM $golang_image AS builder
 WORKDIR /go/src/github.com/kubernetes-sigs/aws-iam-authenticator
 COPY . .
 RUN go version
-RUN goproxy=https://goproxy.io go mod download
+
 ARG TARGETOS TARGETARCH
-RUN GOOS=$TARGETOS GOARCH=$TARGETARCH make bin
+RUN GOPROXY=direct GOOS=$TARGETOS GOARCH=$TARGETARCH make bin
```

Comment:
hopefully this resolves the errors we're from prow after merging to `master` https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/aws-iam-authenticator-push-images/1857485944031023104

(not sure where those images are getting pushed to, or if they should still be getting built and pushed)

![image](https://github.com/user-attachments/assets/0ced9094-98b9-4d25-b3df-9be082f80f99)